pybigquery/sqlalchemy_bigquery.py

```diff
@@ -267,47 +268,61 @@ def __init__(
     def dbapi(cls):
         return dbapi
 
-    def create_connect_args(self, url):
-        location, dataset_id, arraysize, credentials_path, default_query_job_config = parse_url(url)
+    @staticmethod
+    def _add_default_dataset_to_job_config(job_config, project_id, dataset_id):
+        # If dataset_id is set, then we know the job_config isn't None
+        if dataset_id:
+            # If project_id is missing, use defualt project_id
+            if not project_id:
+                _, project_id = auth.default()
 
-        self.arraysize = self.arraysize or arraysize
-        self.location = location or self.location
-        self.credentials_path = credentials_path or self.credentials_path
-        self.dataset_id = dataset_id
+            job_config.default_dataset = f'{project_id}.{dataset_id}'
```

Comment:
What versions of Python does `pybigquery` want to support? If we need to support Python 3.5, I don't think we can use fstrings.

Reply:
Resolved